frontend/components/vault/WithdrawDaiFromVat.vue

```diff
@@ -0,0 +1,116 @@
+            transaction is required to move that DAI to your wallet.
+        </TextBlock>
+        <div class="my-4">
+            <WalletAuthorizationCheckPanel
+                :disabled="!isWalletConnected"
+                :wallet-address="walletAddress"
+                :is-wallet-authorized="isWalletAuthorized"
+                :is-explanations-shown="isExplanationsShown"
+                :is-loading="isAuthorizing"
+                @authorizeWallet="$emit('authorizeWallet')"
+            />
+        </div>
+        <div class="flex justify-end mt-2 gap-5">
+            <BaseButton
+                :disabled="!isWalletConnected || !isWalletAuthorized || !hasDaiToWithdraw || isWithdrawing"
+                @click="$emit('manageVat')"
+            >
+                Withdraw DAI partially
+            </BaseButton>
+            <BaseButton
+                type="primary"
+                :disabled="!isWalletConnected || !isWalletAuthorized || !hasDaiToWithdraw"
+                :is-loading="isWithdrawing"
+                @click="$emit('withdrawAllDaiFromVat')"
+            >
+                <span v-if="hasDaiToWithdraw">
+                    Withdraw <FormatCurrency :value="daiVatBalance" currency="DAI" /> from VAT
+                </span>
+                <span v-else> Nothing to withdraw yet </span>
+            </BaseButton>
+        </div>
+    </BasePanel>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import { VaultTransactionState } from 'auctions-core/src/types';
+import BaseButton from '~/components/common/inputs/BaseButton.vue';
+import TextBlock from '~/components/common/other/TextBlock.vue';
+import FormatCurrency from '~/components/common/formatters/FormatCurrency.vue';
+import BasePanel from '~/components/common/other/BasePanel.vue';
+import WalletAuthorizationCheckPanel from '~/components/panels/WalletAuthorizationCheckPanel.vue';
+
+export default Vue.extend({
+    name: 'WithdrawDAIPanel',
+    components: {
+        BaseButton,
+        BasePanel,
+        TextBlock,
+        FormatCurrency,
+        WalletAuthorizationCheckPanel,
+    },
+    props: {
+        walletAddress: {
+            type: String,
+            default: undefined,
+        },
+        daiVatBalance: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: undefined,
+        },
+        isAuthorizing: {
+            type: Boolean,
+            default: false,
+        },
+        isWalletAuthorized: {
+            type: Boolean,
+            default: false,
+        },
+        isWithdrawing: {
+            type: Boolean,
+            default: false,
+        },
+        isExplanationsShown: {
+            type: Boolean,
+            default: true,
+        },
+        vaultState: {
+            type: String as Vue.PropType<VaultTransactionState>,
+            default: 'liquidatable',
+        },
+    },
+    computed: {
+        currentStateAndTitle(): PanelProps {
```

Comment:
You also have no state here what to display if no wallet is connected! 

This related to my other two points: Will we ever show this panel in a state if no wallet is connected?

YES: Keep the props optional and add a state here to handle it
NO: Make the props above required, to ensure there wont be an issue here!